lib/bap_disasm/bap_disasm_rec.ml

```diff
@@ -66,121 +63,38 @@ module Cfg = Graphlib.Make(Node)(Edge)
 
 type cfg = Cfg.t [@@deriving compare]
 
-
-(** Interval tree spanning visited memory *)
-module type Span = sig
-  type range = addr * addr
-  type t [@@deriving sexp_of]
-
-  val empty : t
-  val add : t -> range -> t
-  val mem : t -> addr -> bool
-  val upper_bound : t -> addr -> addr option
-  val min : t -> addr option
-  val max : t -> addr option
-  val pp : Format.formatter -> t -> unit
+module Visited = struct
```

Comment:
Why is visited an addr->addr mapping, not just a set? What is the payload? 
Why you're not using `Bap.Std.Memmap` or `Bap.Std.Table`?

Reply:
Set is not sufficient, because we need to know both first and last addresses of an instruction to figure out if it's a block terminator or not. `Bap.Std.Table` doesn't fit for this particular task because we have intersected memory regions. We can also use `Bap.Std.Memmap` in most of the cases, but it won't be so convenient as just the `addr->addr` mapping since we use addresses everywhere as a key for tables. And sometimes it will be even hard to use `mem` as a key when we need to infer the next key from a given one, which is easy to do in case of address by incrementing it, and not so easy for memory chunk.